src/app/GitCommands/Git/GitVersion.cs

```diff
@@ -1,182 +1,200 @@
+    public static readonly GitVersion LastRecommendedVersion = new("2.46.0");
+
+    /// <summary>
+    /// The oldest version with reasonable reliable support in GE.
+    /// Older than this version is red in settings.
+    /// </summary>
+    public static readonly GitVersion LastSupportedVersion = new("2.25.0");
+
+    /// <summary>
+    /// The oldest Git version without known incompatibilities.
+    /// </summary>
+    public static readonly GitVersion LastVersionWithoutKnownLimitations = new("2.15.2");
+
+    private static readonly Dictionary<string, GitVersion> _current = [];
+
+    /// <summary>
+    /// GitVersion for the native ("Windows") Git
+    /// </summary>
+    public static IGitVersion Current { get; private set; } = CurrentVersion();
+
+    /// <summary>
+    /// Gets the Git version for the <paramref name="gitExecutable"/>.
+    /// </summary>
+    /// <returns>A <see cref="GitVersion"/>.</returns>
+    public static IGitVersion CurrentVersion(IExecutable? gitExecutable = null)
     {
-        private static readonly GitVersion v2_19_0 = new("2.19.0");
-        private static readonly GitVersion v2_20_0 = new("2.20.0");
-        private static readonly GitVersion v2_32_0 = new("2.32.0");
-        private static readonly GitVersion v2_35_0 = new("2.35.0");
-        private static readonly GitVersion v2_38_0 = new("2.38.0");
-
-        /// <summary>
-        /// The recommended Git version (normally latest official before a GE release).
-        /// This and later versions are green in the settings check.
-        /// </summary>
-        public static readonly GitVersion LastRecommendedVersion = new("2.46.0");
-
-        /// <summary>
-        /// The oldest version with reasonable reliable support in GE.
-        /// Older than this version is red in settings.
-        /// </summary>
-        public static readonly GitVersion LastSupportedVersion = new("2.25.0");
-
-        /// <summary>
-        /// The oldest Git version without known incompatibilities.
-        /// </summary>
-        public static readonly GitVersion LastVersionWithoutKnownLimitations = new("2.15.2");
-
-        private static readonly Dictionary<string, GitVersion> _current = [];
-
-        /// <summary>
-        /// GitVersion for the native ("Windows") Git
-        /// </summary>
-        public static IGitVersion Current => CurrentVersion();
-
-        /// <summary>
-        /// The GitVersion for the gitIdentifiable
-        /// </summary>
-        /// <param name="gitIdentifiable">The unique identification of the Git executable</param>
-        /// <returns>The GitVersion</returns>
-        public static IGitVersion CurrentVersion(IExecutable gitExec = null, string gitIdentifiable = "")
+        string gitIdentifiable
+            = gitExecutable is null ? AppSettings.GitCommand
+            : string.IsNullOrWhiteSpace(gitExecutable.PrefixArguments) ? gitExecutable.Command
+            : $"{gitExecutable.Command} {gitExecutable.PrefixArguments}";
```

Comment:
please reformat or indent
```suggestion
            = gitExecutable is null
                ? AppSettings.GitCommand
                : string.IsNullOrWhiteSpace(gitExecutable.PrefixArguments) 
                   ? gitExecutable.Command
                   : $"{gitExecutable.Command} {gitExecutable.PrefixArguments}";
```